core/unit_test/tools/TestLogicalSpaces.hpp

```diff
@@ -189,14 +189,5 @@ TEST(defaultdevicetype, chained_logical_spaces) { test_chained_spaces(); }
 TEST(defaultdevicetype, access_allowed) {
   test_allowed_access<fake_memory_space>();
 }
-// FIXME_SYCL
-#if !(defined(KOKKOS_COMPILER_INTEL) && defined(KOKKOS_ENABLE_SYCL))
-TEST(defaultdevicetype_DeathTest, access_forbidden) {
-  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
-  ASSERT_DEATH(
-      { test_allowed_access<semantically_independent_logical_space>(); },
-      "Kokkos::View ERROR: attempt to access inaccessible memory space");
-}
```

Comment:
Why is it ok to remove? 
Then the `semantically_independent_logical_space` alias is never used.

Reply:
Also how is int related to this PR?

Reply:
It fails the `static_assert` and thus doesn't even compile. We could replace with a compile check for accessibility instead.

Reply:
This test is about memory access violation.  Why are we not able to build the view anymore?
It is out of the question to merge that change without going to the bottom of this.

Reply:
This test tries to create a View in a memory space that is not accessible from its execution space and then tries to access it in a `parallel_for` using that memory space. We expect an access error that is currently happening in the `parallel_for` although we initialize the view. It doesn't already fail in the View initialization because we are using `memset` which doesn't check for memory access (I have been running into (non-Kokkos) Cuda runtime errors because of that). With the original version of this patch, we don't even allow the View initialization to happen which I think is the right thing to do here. To still keep the test, I am now using an unmanaged view.

In the end, the open question still is "Should we use the static_assert only if the View is actually initialized?".
If yes, it would be sufficient to use `Kokkos::WithoutInitializing` instead of an unmanaged View.